compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -3475,19 +3475,15 @@ public BLangConstant transformEnumMember(EnumMemberNode member, Boolean publicQu
 
         bLangConstant.setName((BLangIdentifier) transform(member.identifier()));
 
-        BLangLiteral literal;
         BLangLiteral deepLiteral;
         if (member.constExprNode().isPresent()) {
-            literal = createSimpleLiteral(member.constExprNode().orElse(null));
+            BLangExpression expression = createExpression(member.constExprNode().orElse(null));
+            bLangConstant.setInitialExpression(expression);
             deepLiteral = createSimpleLiteral(member.constExprNode().orElse(null));
```

Comment:
This should also be replaced right?

Reply:
Yes. However, if we do so, how can we rewrite the below logic which occurs in the same method:
```
        if (deepLiteral.originalValue != "") {
            BLangFiniteTypeNode typeNodeAssosiated = (BLangFiniteTypeNode) TreeBuilder.createFiniteTypeNode();
            deepLiteral.originalValue = null;
            typeNodeAssosiated.addValue(deepLiteral);
            bLangConstant.associatedTypeDefinition = createTypeDefinitionWithTypeNode(typeNodeAssosiated);
        } else {
            bLangConstant.associatedTypeDefinition = null;
        }
```